### PR TITLE
AWS FSx integration

### PR DESCRIPTION
**Pull request by hlepesant**
Hi, 
For customer's needs I had to create this integration.

### CODE REVIEW

File: modules/integration_aws-fsx/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+## How to use this module?
+
+This directory defines a [Terraform](https://www.terraform.io/)
+[module](https://www.terraform.io/language/modules/syntax) you can use in your
+existing [stack](https://github.com/claranet/terraform-signalfx-detectors/wiki/Getting-started#stack) by adding a
+`module` configuration and setting its `source` parameter to URL of this folder:
+
+```hcl
+module "signalfx-detectors-integration-aws-fsx" {
+  source = "github.com/hlepesant/terraform-signalfx-detectors.git//modules/integration_aws-fsx?ref={revision}"
```
Comment (pdecat) [inline modules/integration_aws-fsx/README.md]:
```suggestion
  source = "github.com/claranet/terraform-signalfx-detectors.git//modules/integration_aws-fsx?ref={revision}"
```

File: modules/integration_aws-fsx/conf/readme.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+documentations:
+  - name: CloudWatch metrics
+    url: 'https://docs.aws.amazon.com/fsx/latest/WindowsGuide/fsx-windows-metrics.html'
+  - name: CloudWatch guide
+    url: 'https://docs.aws.amazon.com/fr_fr/fsx/latest/WindowsGuide/monitoring-cloudwatch.html'
```
Comment (pdecat) [inline modules/integration_aws-fsx/conf/readme.yaml:5]: Does this module only apply to FSx for Windows?

If not, maybe also link the other variants:

* https://docs.aws.amazon.com/fsx/latest/LustreGuide/monitoring-cloudwatch.html
* https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/monitoring-cloudwatch.html / https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/file-system-metrics.html / https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/volume-metrics.html
* https://docs.aws.amazon.com/fsx/latest/OpenZFSGuide/monitoring-cloudwatch.html

Comment (pdecat) [reply modules/integration_aws-fsx/conf/readme.yaml:5]: Also, link to original, non translated, version:
```suggestion
  - name: CloudWatch metrics
    url: 'https://docs.aws.amazon.com/fsx/latest/WindowsGuide/fsx-windows-metrics.html'
  - name: CloudWatch guide
    url: 'https://docs.aws.amazon.com/fsx/latest/WindowsGuide/monitoring-cloudwatch.html'
```

File: modules/integration_aws-fsx/conf/01-used-space.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+filtering: "filter('namespace', 'AWS/FSx')"
+value_unit: "Gibibyte"
+transformation: ".max(over='15m')"
+signals:
+  used_space:
+    metric: "FreeStorageCapacity"
+    filter: "filter('stat', 'mean')"
+  signal:
+    formula:
+      used_space.scale(1/1024**3)
```
Comment (pdecat) [inline modules/integration_aws-fsx/conf/01-used-space.yaml:12]: How about renaming this signal and the file to match the metric name:
```suggestion
  free_space:
    metric: "FreeStorageCapacity"
    filter: "filter('stat', 'mean')"
  signal:
    formula:
      free_space.scale(1/1024**3)
```

File: modules/integration_aws-fsx/conf/01-used-space.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+rules:
+  critical:
+    comparator: "<"
+  major:
+    comparator: "<"
+    dependency: critical
```
Comment (pdecat) [inline modules/integration_aws-fsx/conf/01-used-space.yaml:18]: Missing thresholds?
```suggestion
  critical:
    comparator: "<"
    threshold: 10
  major:
    comparator: "<"
    dependency: critical
    threshold: 20
```

File: modules/integration_aws-fsx/README.md (hunk 1 above)
Comment (hlepesant) [reply modules/integration_aws-fsx/README.md]: fixed

File: modules/integration_aws-fsx/conf/readme.yaml (hunk 2 above)
Comment (hlepesant) [reply modules/integration_aws-fsx/conf/readme.yaml:5]: Unfortunately the FSx variants do not have the same metrics.
May be Il should rename the module's directory to `modules/integration_aws-fsx-windows-file-server/`

File: modules/integration_aws-fsx/conf/01-used-space.yaml (hunk 3 above)
Comment (hlepesant) [reply modules/integration_aws-fsx/conf/01-used-space.yaml:12]: done

File: modules/integration_aws-fsx/conf/01-used-space.yaml (hunk 4 above)
Comment (hlepesant) [reply modules/integration_aws-fsx/conf/01-used-space.yaml:18]: done

File: modules/integration_aws-fsx/conf/readme.yaml (hunk 2 above)
Comment (pdecat) [reply modules/integration_aws-fsx/conf/readme.yaml]: If so, please rename it indeed!

File: modules/integration_aws-fsx-windows-file-server/README.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
+# AWS-FSX SignalFx detectors
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/README.md:1]:
```suggestion
# AWS FSx for Windows File Server SignalFx detectors
```

File: modules/integration_aws-fsx-windows-file-server/README.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+## How to use this module?
+
+This directory defines a [Terraform](https://www.terraform.io/)
+[module](https://www.terraform.io/language/modules/syntax) you can use in your
+existing [stack](https://github.com/claranet/terraform-signalfx-detectors/wiki/Getting-started#stack) by adding a
+`module` configuration and setting its `source` parameter to URL of this folder:
+
+```hcl
+module "signalfx-detectors-integration-aws-fsx" {
+  source = "github.com/hlepesant/terraform-signalfx-detectors.git//modules/integration_aws-fsx?ref={revision}"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/README.md:27]:
```suggestion
  source = "github.com/claranet/terraform-signalfx-detectors.git//modules/integration_aws-fsx-windows-file-server?ref={revision}"
```

File: modules/integration_aws-fsx-windows-file-server/conf/00-heartbeat.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/00-heartbeat.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server

File: modules/integration_aws-fsx-windows-file-server/conf/01-free-storage-capacity.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/01-free-storage-capacity.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server"
```

File: modules/integration_aws-fsx-windows-file-server/conf/02-cpu-utilization.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/02-cpu-utilization.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server"
```

File: modules/integration_aws-fsx-windows-file-server/conf/03-memory-utilization.yaml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/03-memory-utilization.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server"
```

File: modules/integration_aws-fsx-windows-file-server/conf/04-network-throughput-utilization.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/04-network-throughput-utilization.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server"
```

File: modules/integration_aws-fsx-windows-file-server/conf/05-fileserver-disk-throughput-utilization.yaml [hunk 12]
```diff
@@ -0,0 +1,17 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/05-fileserver-disk-throughput-utilization.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server"
```

File: modules/integration_aws-fsx-windows-file-server/conf/06-storage-capacity-utilization.yaml [hunk 13]
```diff
@@ -0,0 +1,17 @@
+module: "AWS FSx"
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/06-storage-capacity-utilization.yaml:1]:
```suggestion
module: "AWS FSx for Windows File Server"
```

File: modules/integration_aws-fsx-windows-file-server/conf/readme.yaml [hunk 14]
```diff
@@ -0,0 +1,26 @@
+documentations:
+  - name: CloudWatch metrics
+    url: 'https://docs.aws.amazon.com/fsx/latest/WindowsGuide/fsx-windows-metrics.html'
+  - name: Amazon FSx for Lustre CloudWatch guide
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/readme.yaml:4]:
```suggestion
  - name: AWS FSx for Windows File Server CloudWatch guide
```

File: modules/integration_aws-rds-common/conf/05-dbload.yml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+rules:
+  major:
+    threshold: 80
+    comparator: ">"
+    lasting_duration: '5m'
+  critical:
+    threshold: 90
+    comparator: ">"
+    lasting_duration: '5m'
+    dependency: major
```
Comment (pdecat) [inline modules/integration_aws-rds-common/conf/05-dbload.yml:22]: This should not be in this PR.

**Review by pdecat — CHANGES_REQUESTED** [review]

File: modules/integration_aws-fsx-windows-file-server/README.md [hunk 16]
```diff
@@ -0,0 +1,146 @@
+# AWS FSx for windows file server SignalFx detectors
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/README.md:1]:
```suggestion
# AWS FSx for Windows File Server SignalFx detectors
```

File: modules/integration_aws-fsx-windows-file-server/conf/01-free-storage-capacity.yaml [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+  free_space:
+    metric: "FreeStorageCapacity"
+    filter: "filter('stat', 'mean')"
+  signal:
+    formula:
+      free_space.scale(1/1024**3)
+rules:
+  critical:
+    comparator: "<"
+    # threshold: 10
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/01-free-storage-capacity.yaml:17]: Why is this commented?

File: modules/integration_aws-fsx-windows-file-server/conf/01-free-storage-capacity.yaml [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+  major:
+    comparator: "<"
+    dependency: critical
+    # threshold: 20
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/conf/01-free-storage-capacity.yaml:21]: Why is this commented?

File: modules/integration_aws-fsx-windows-file-server/tags.tf [hunk 19]
```diff
@@ -0,0 +1,4 @@
+locals {
+  tags = ["integration", "aws-fsx"]
```
Comment (pdecat) [inline modules/integration_aws-fsx-windows-file-server/tags.tf:2]: How about this?
```suggestion
  tags = ["integration", "aws-fsx-windows"]
```

**Review by pdecat — CHANGES_REQUESTED** [review]

**Review by pdecat — APPROVED** [review]
LGTM!

**Review by haedri — CHANGES_REQUESTED** [review]
Could you `make update-module-doc` this module ? it seems the title is incorrect in README.md

**Comment by hlepesant** [comment]
make update-module-doc have been done

**Pull request closed without merging**